packages/experimental-package/src/components/PageHeader/ActionBarItem.js

```diff
@@ -6,72 +6,48 @@
 //
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import cx from 'classnames';
 
-// import { settings } from 'carbon-components';
-// const { prefix } = settings;
-
 import { expPrefix } from '../../global/js/settings';
 
 import { Button } from 'carbon-components-react';
 
 const blockClass = `${expPrefix}-action-bar-item`;
 
-export const ActionBarItem = ({
-  className,
-  disabled,
-  icon,
-  label,
-  onClick,
-}) => {
+export const ActionBarItem = (props) => {
+  const className = cx([blockClass, props.className]);
+
   return (
     <Button
-      className={cx([blockClass, className])}
-      disabled={disabled}
-      hasIconOnly
-      iconDescription={label}
-      kind="ghost"
-      onClick={onClick}
-      renderIcon={icon}
-      size="field"
-      tooltipPosition="bottom"
-      tooltipAlignment="end"
-      type="button"></Button>
+      {...{
+        ...props,
+        className,
+        hasIconOnly: true,
+        kind: 'ghost',
+        size: 'field',
+        tooltipPosition: 'bottom',
+        tooltipAlignment: 'end',
+        type: 'button',
+      }}></Button>
   );
 };
 
-ActionBarItem.propTypes = {
-  /**
-   * Specifies class(es) to be applied to the top-level PageHeader node.
-   * Optional.
-   */
-  className: PropTypes.string,
-  /**
-   * Specifies whether the ActionBarItem should be disabled, or not.
-   * Optional.
-   */
-  disabled: PropTypes.bool,
-  /**
-   * The icon for the ActionBarItem.
-   * Mandatory.
-   */
-  icon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
-  /**
-   * A label to describe the ActionBarItem. Displayed on hover and used as
-   * assistive text.
-   * Mandatory.
-   */
-  label: PropTypes.string.isRequired,
-  /**
-   * Specifies an onClick handler that is called when the ActionBarItem is
-   * clicked. Optional.
-   */
-  onClick: PropTypes.func,
-};
-
-ActionBarItem.defaultProps = {
-  className: '',
-  disabled: false,
-};
+const reservedProps = [
+  'hasIconOnly',
+  'kind',
+  'size',
+  'tooltipPosition',
+  'tooltipAlignment',
+  'type',
+];
+const propTypes = { ...Button.propTypes };
+const defaultProps = { ...Button.defaultProps };
+
+reservedProps.forEach((prop) => {
```

Comment:
i'm not sure i'm understanding what the intent is here. is this to make sure that certain props passed from the user to `ActionBarItem` don't make it to the underlying `Button` component?

Reply:
Yes.

Reply:
hmmm interesting. i don't believe i've encountered this pattern before. so you're just copying the same prop types and default props from `Button` to `ActionBarItem` and then removing props that we don't want the user to pass. i'm a little torn on this. i think this is an interesting way to cut down on redundant code, but I think this hurts the optics. if i'm trying to find out what the proptypes are for this component I don't think I'd want to have to go to a completely separate repo to find them. i mean i guess you can find this information from the storybook, but code wise I think it hurts readability.

i think i'd like to see some other peoples opinion on this pattern.

Reply:
Me either it does have some benefits in that it reduces the components exposure to changes in Carbon. It should even carry on working even if PropTypes and storybook become out of sync with Carbon.

It is frustrating that it's necessary to declare any of the PropTypes, they're only required for storybook.

Reply:
@davidmenendez can we move forward with this and discuss wrapping in #138 ?